android/app/src/main/java/com/emergetools/hackernews/features/comments/CommentsDomain.kt

```diff
@@ -229,17 +249,34 @@ class CommentsViewModel(
         viewModelScope.launch {
           val current = internalState.value
           if (current is CommentsState.Content && current.postComment != null) {
+            val (parentId, gotoUrl, hmac, _, text) = current.postComment
+            val updatedComments = webClient.postComment(parentId, gotoUrl, hmac, text)
             internalState.value = current.copy(
-              commentText = ""
+              commentText = "",
+              comments = updatedComments.map { it.toCommentState() }
             )
-            val (parentId, gotoUrl, hmac, _, text) = current.postComment
-            webClient.postComment(parentId, gotoUrl, hmac, text)
           }
         }
       }
     }
   }
 
+  private fun CommentInfo.toCommentState(): CommentState.Content {
+    return CommentState.Content(
+      id = id,
+      author = user,
+      content = text,
+      timeLabel = relativeTimeStamp(
+        epochSeconds = LocalDateTime.parse(age)
```

Comment:
Not sure if this is shared with posts as well, but could be good to unify/reuse from a util or extension fun.

Reply:
Yea it's fairly similar. In fact comments and posts are the same "type" technically, everything is an Item. I'll have a cleanup round where hopefully I can simplify the data models a lot.